feat(ui): UI flair, better themes & settings, font support

davo0411 opened the pull request:
KEPT AS TO-DO & TRACKING LIST FOR UI CHANGES, I WILL DELETE WHEN DONE.

PR 1:
- [x] Util and functions, Theme saving to .json system. (NOT INCLUDING ANY THEMES.jsons), expanded font system. Only Jost.
Settings UI Overhaul & new UI settings (Border controls, scrollbar controls.)

PR 2:
- [x] Changes to Perf overlay removing dropdowns from overlay.

PR 3: 
- [ ] Background Blur, toggles, other UI candy i added.

PR 4:
- [ ] Add other fonts, other themes

davo0411 committed:
v1

davo0411 committed:
Theme multi choosing

davo0411 committed:
Updates & Fine Tuning

davo0411 committed:
Save settings fixes

davo0411 committed:
Font support, ctd on changing fonts.

davo0411 committed:
font fixes

davo0411 committed:
fixed font rendering

davo0411 committed:
Fixes

davo0411 committed:
Fixes x2

davo0411 committed:
Update Menu.cpp

davo0411 committed:
Update Default.json

davo0411 committed:
Update Menu.h

davo0411 committed:
Merge branch 'dev' into separate-menu-settings

pre-commit-ci[bot] committed:
style: 🎨 apply pre-commit.ci formatting

Automated formatting by clang-format, prettier, and other hooks.
See https://pre-commit.ci for details.

davo0411 committed:
build fixes

pre-commit-ci[bot] committed:
style: 🎨 apply pre-commit.ci formatting

Automated formatting by clang-format, prettier, and other hooks.
See https://pre-commit.ci for details.

davo0411 committed:
fixes

davo0411 committed:
RAAAAHHHHH IT WORKS

davo0411 committed:
Update Default.json

davo0411 committed:
background blur added to themes

davo0411 committed:
Fixes, Credits

davo0411 committed:
border settings

pre-commit-ci[bot] committed:
style: 🎨 apply pre-commit.ci formatting

Automated formatting by clang-format, prettier, and other hooks.
See https://pre-commit.ci for details.

davo0411 committed:
Font Family Support

davo0411 committed:
font updates

davo0411 committed:
fixes

pre-commit-ci[bot] committed:
style: 🎨 apply pre-commit.ci formatting

Automated formatting by clang-format, prettier, and other hooks.
See https://pre-commit.ci for details.

davo0411 committed:
ai recommended fixes

pre-commit-ci[bot] committed:
style: 🎨 apply pre-commit.ci formatting

Automated formatting by clang-format, prettier, and other hooks.
See https://pre-commit.ci for details.

alandtse reviewed (changes requested):
If this is just themes, why does it include a blur shader? Please break this up into logical additions. It can include helper functions and touch lots of files, but there seems to be hidden functionality in here not described.

davo0411 committed:
Merge branch 'separate-menu-settings' into menu-settings-updated

davo0411 committed:
Merge pull request #15 from davo0411/menu-settings-updated

update to dev

davo0411 committed:
Implement collapsible sections in PerformanceOverlay

Added collapsible sections for FPS, Draw Calls, VRAM, and A/B Test results in the Performance Overlay. Updated function signatures and logic to support the new collapsible behavior.
